src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/20240802123234_Routine_RebuildIndexes.sql

```diff
@@ -0,0 +1,283 @@
+          __RebuildIndexesAlterIndexes.Info           = IIF(@FragPercentBefore >= @FragmentationThresholdRebuild,
+                                                            'Rebuilt', 'Reorganized')
+      FROM (SELECT ips.avg_fragmentation_in_percent AS FragPercent
+            FROM #TableList AS tl
+                   JOIN sys.dm_db_index_physical_stats(DB_ID(), NULL, NULL, NULL, 'LIMITED') ips
+                        ON ips.object_id = OBJECT_ID(tl.ObjectName, 'U')
+                   JOIN sys.indexes idx ON idx.object_id = ips.object_id AND idx.index_id = ips.index_id
+                   JOIN sys.objects obj ON obj.object_id = ips.object_id
+                   JOIN sys.schemas sch ON sch.schema_id = obj.schema_id
+            WHERE ips.alloc_unit_type_desc = 'IN_ROW_DATA'
+              AND tl.ObjectName = @ObjectName
+              AND sch.Name = @SchemaName
+              AND idx.name = @IndexName) AS Fragmented
+      WHERE __RebuildIndexesAlterIndexes.RunId = @RunId
+        AND __RebuildIndexesAlterIndexes.Id = @StatsRowIndex;
+
+      IF @StopTime < GETUTCDATE()
+        BEGIN
+          -- Update Info for all unprocessed indexes, because we're skipping them now we've hit @StopTime
+          UPDATE __RebuildIndexesAlterIndexes
+          SET Info = 'Skipped: stored procedure run time exceeded timeout'
+          WHERE Info IS NULL;
+
+          SET @SkipRemainingTasks = 1;
+        END
+
+      IF @StatsRowIndex >= @StatsCount
+        RAISERROR ('Completed optimising indexes for tables.', 0, 1, NULL) WITH NOWAIT;
+
+      SET @StatsRowIndex += 1;
+    END
+
+  /*-----------------------------------------------------------------------*/
+
+  SET @TablesRowIndex = 1;
+  SELECT @TablesCount = COUNT(Id) FROM #TableList;
+  WHILE @TablesRowIndex <= @TablesCount AND @SkipRemainingTasks = 0
+    BEGIN
+      DECLARE @CurrentTable NVARCHAR(MAX) = '';
+      SELECT @CurrentTable = ObjectName
+      FROM #TableList
+      WHERE Id = @TablesRowIndex;
+
+      SET @Command = 'UPDATE STATISTICS ' + @CurrentTable;
+
+      -- Save start time
+      UPDATE __RebuildIndexesUpdateStatistics
+      SET StartTime = GETUTCDATE()
+      WHERE RunId = @RunId
+        AND TableName = @CurrentTable;
+
+      -- Run
+      RAISERROR ('Updating statistics on table ''%s''.', 0, 1, @CurrentTable) WITH NOWAIT;
+      EXEC (@Command);
+
+      -- Save end time
+      UPDATE __RebuildIndexesUpdateStatistics
+      SET EndTime = GETUTCDATE()
+      WHERE RunId = @RunId
+        AND TableName = @CurrentTable;
+
+      -- If we completed the last update statistics query, it doesn't matter if we've exceeded @StopTime: we've not skipped anything
+      IF @TablesRowIndex + 1 <= @TablesCount AND @StopTime < GETUTCDATE()
+        SET @SkipRemainingTasks = 1;
+
+      IF @TablesRowIndex + 1 > @TablesCount
+        RAISERROR ('Completed updating statistics on tables.', 0, 1, NULL) WITH NOWAIT;
```

Comment:
Could move this outside the `WHILE` without the check on count to become a log message to signal that this stage has completed even if something has been skipped.